### PR TITLE
Added support for user defined custom validation callback

### PR DESCRIPTION
**Pull request by davidstevens37**
This change adds the ability for a user to define a callback to test the property against whatever they want. 

the callback is passed 3 arguments: key, value, and the model's scope. This is awesome because you're able to compare against another property (or properties) on the model. return a truthy value back and it will pass the validation or return a falsy value back to fail the validation. the second example works the exact same way except will use the custom message instead of default.

if the object syntax (2) is used, and the user does not define a `validate` property or the `validate` property is not a function, the validator will not invoke the callback and the validation will pass.

1)
````js
  validations: {
    lotteryNumber: {
      custom: funciton(key, value, model){
        return model.get('accountBalance') > 1 ? true : false;
      }
    }
  }
````

2)
````js
  validations: {
    lotteryNumber: {
      custom: {
        validation: funciton(key, value, model){
          return model.get('accountBalance') > 1 ? true : false;
        },
        message: 'You can't win off of good looks and charm.'
      }
    }
  }
````

### CODE REVIEW

File: addon/mixins/model-validator.js [hunk 1]
```diff
@@ -237,6 +247,13 @@ export default Ember.Mixin.create({
     if(options.hasOwnProperty('only') && options.only.indexOf(property) === -1){ validateThis = false; }
     return validateThis;
   },
+  _getCustomValidator: function(validation){
+    var customValidator = validation.custom;
+    if (Ember.typeOf(validation.custom) === 'object' && validation.custom.hasOwnProperty('validation')) {
+      customValidator = validation.custom.validation;
+    } 
+    return typeof customValidator === 'function' ? customValidator : false;
```
Comment (esbanarango) [inline addon/mixins/model-validator.js]: @davidstevens37 Shouldn't this return the return value from the function?

````js
return typeof customValidator === 'function' ? customValidator() : false;
````

File: README.md [hunk 2]
```diff
@@ -223,6 +224,34 @@ The value has to have only numeric values.
   }
 ````
 
+### CustomValidation
+Define a custom callback funciton to validate the model's value. The validation callback is passed 3 values: the key, value, model's scope. return true (or a truthy value) to pass the validation, return false (or falsy value) to fail the validation.
+
+````js
+  validations: {
+    lotteryNumber: {
+      custom: funciton(key, value, model){
```
Comment (esbanarango) [inline README.md]: @davidstevens37 Why do we need these three values?

File: addon/mixins/model-validator.js (hunk 1 above)
Comment (davidstevens37) [inline addon/mixins/model-validator.js]: this is determining whether the customValidation function is located at `validations.{property}.custom` or `validations.{property}.custom.validation` (in the case that a custom message is defined), ensuring that the callback is in fact a function and returning the function to be invoked by the `_validateCustom` method on line 52.

File: README.md (hunk 2 above)
Comment (davidstevens37) [inline README.md]: passing the `key` and `value` make validating the value less verbose and reusable if many properties share common custom validation. the 3rd param is important to access other properties on the model, this way we can validate properties that have dependencies on other properties. Im having a hard time coming up with an example off the cuff, but for example: a user's password cannot be equal to their username.

```js
validations: {
  password: {
    custom: function (key, value, _this) { // var key = 'password', var value = 'MyPassword123', _this = (the model's 'this')
      var password = value;
      var username = _this.get('username');
      return (password !== username);
    }
  }
}
```

**Comment by esbanarango** [comment]
Merging! Thank you @davidstevens37! Great work.

**Pull request merged**